QuickFIXn/Session.cs

```diff
@@ -630,19 +660,19 @@ internal void Next(MessageBuilder msgBuilder)
                 }
                 else
                 {
-                    Log.OnEvent(uvx.ToString());
+                    Log.Log(LogLevel.Error, uvx, "{Message}", uvx.ToString());
                     GenerateLogout(uvx.Message);
                     _state.IncrNextTargetMsgSeqNum();
                 }
             }
             catch (UnsupportedMessageType e)
             {
-                Log.OnEvent("Unsupported message type: " + e.Message);
+                Log.Log(LogLevel.Error, e, "Unsupported message type: {Message}", e.Message);
                 GenerateBusinessMessageReject(message!, Fields.BusinessRejectReason.UNKNOWN_MESSAGE_TYPE, 0);
             }
             catch (FieldNotFoundException e)
             {
-                Log.OnEvent("Rejecting invalid message, field not found: " + e.Message);
+                Log.Log(LogLevel.Information, e, "Rejecting invalid message, field not found: {Message}", e.Message);
```

Comment:
Is the `{Message}` part redundant?

Reply:
These probably are, but I'm going to leave them in anyway (and change more log calls to be like this), because it's nice to have the actual error message in the first line.

After seeing this "in the wild" for a bit, I may come reevaluate this decision.